src/store/publicChannels/publicChannels.slice.ts

```diff
@@ -3,54 +3,78 @@ import { createSlice, EntityState, PayloadAction } from '@reduxjs/toolkit';
 import { StoreKeys } from '../store.keys';
 
 import {
-  publicChannelMessagesAdapter,
+  publicChannelMessagesIdsAdapter,
   publicChannelsAdapter,
 } from './publicChannels.adapter';
-import { IMessage, IChannelInfo } from './publicChannels.types';
+import { IChannelInfo, IMessage } from './publicChannels.types';
 
 export class PublicChannelsState {
-  public currentChannelAddress: string = '';
   public channels: EntityState<IChannelInfo> =
     publicChannelsAdapter.getInitialState();
-  public messages: EntityState<IMessage> =
-    publicChannelMessagesAdapter.getInitialState();
+
+  public currentChannelMessagesIds: EntityState<string> =
+    publicChannelMessagesIdsAdapter.getInitialState();
+
+  public currentChannelMessages: ChannelMessages = {};
 }
 
-export interface ChannelInfoResponse {
+export interface GetPublicChannelsResponse {
   [name: string]: IChannelInfo;
 }
 
+export interface ChannelMessages {
+  [id: string]: IMessage;
+}
+
+export interface ChannelMessagesIdsResponse {
+  channelAddress: string;
+  ids: string[];
+}
+
+export interface AskForMessagesPayload {
+  channelAddress: string;
+  ids: string[];
+}
+
+export interface AskForMessagesResponse {
+  channelAddress: string;
+  messages: IMessage[];
+}
+
 export const publicChannelsSlice = createSlice({
   initialState: { ...new PublicChannelsState() },
   name: StoreKeys.PublicChannels,
   reducers: {
-    fetchAllMessages: (state, _action: PayloadAction<string>) => state,
-    responseFetchAllMessages: (
-      state,
-      action: PayloadAction<{
-        channelAddress: string;
-        messages: IMessage[];
-      }>,
-    ) => {
-      publicChannelMessagesAdapter.setAll(
-        state.messages,
-        action.payload.messages,
-      );
-    },
     getPublicChannels: state => state,
-    setCurrentChannelAddress: (state, action: PayloadAction<string>) => {
-      state.currentChannelAddress = action.payload;
-    },
     responseGetPublicChannels: (
       state,
-      action: PayloadAction<ChannelInfoResponse>,
+      action: PayloadAction<GetPublicChannelsResponse>,
     ) => {
       publicChannelsAdapter.setAll(
         state.channels,
         Object.values(action.payload),
       );
     },
     subscribeForTopic: (state, _action: PayloadAction<IChannelInfo>) => state,
+    responseSendIds: (
+      state,
+      action: PayloadAction<ChannelMessagesIdsResponse>,
+    ) => {
+      publicChannelMessagesIdsAdapter.setAll(
```

Comment:
I would not use entityAdapter for messageIds. 
You only ever replace all of them. I think I would use a simple `string[]` for that